storage-api/src/java/org/apache/hadoop/hive/common/type/CalendarUtils.java

```diff
@@ -161,6 +161,27 @@ public static long convertTimeToProleptic(long hybrid) {
     return proleptic;
   }
 
+  /**
+   * Convert epoch microseconds from the hybrid Julian/Gregorian calendar to the
+   * proleptic Gregorian.
+   * @param hybridMicros Microseconds of epoch in the hybrid Julian/Gregorian
+   * @return Microseconds of epoch in the proleptic Gregorian
+   */
+  public static long convertTimeToProlepticMicros(long hybridMicros) {
+    long prolepticMicros = hybridMicros;
+    long hybridMillis = hybridMicros / 1_000L; // Convert micros to millis
```

Comment:
This has the effect of removing the precision of the micros by a factor of 1000. Consider using micros itself.
Can we create SWITCHOVER_MICROS using the same logic as SWITCHOVER_MILLIS for this?